libs/libarchfpga/src/physical_types.h

```diff
@@ -413,6 +415,53 @@ struct t_grid_def {
     std::vector<t_layer_def> layers;
 };
 
+/************************* VIB_GRID ***********************************/
+/* Describe different VIB type on different locations by immitating t_grid_loc_def. */
+
+struct t_vib_grid_loc_def {
+    t_vib_grid_loc_def(std::string block_type_val, int priority_val)
+        : block_type(block_type_val)
+        , priority(priority_val)
+        , x("0", "W-1", "max(w+1,W)", "w") //Fill in x direction, no repeat, incr by block width
+        , y("0", "H-1", "max(h+1,H)", "h") //Fill in y direction, no repeat, incr by block height
+    {}
+
+    std::string block_type; //The block type name
+
+    int priority = 0; //Priority of the specification.
+                      // In case of conflicting specifications
+                      // the largest priority wins.
+
+    t_grid_loc_spec x; //Horizontal location specification
+    t_grid_loc_spec y; //Veritcal location specification
+
+    // When 1 metadata tag is split among multiple t_grid_loc_def, one
+    // t_grid_loc_def is arbitrarily chosen to own the metadata, and the other
+    // t_grid_loc_def point to the owned version.
+    std::unique_ptr<t_metadata_dict> owned_meta;
```

Comment:
If the metadata is not used for now, we do not need to include them in the VIB data structure. 
Since our objective is to separate the codes to the most (avoid merging conflicts as much as possible), I suggest to decouple with the existing data structure ``t_grid_loc_spec`` and ``enum GridDefType``. As a result, all the VIB-related data structure can be moved to the ``vib_inf.[h|cpp]``

Reply:
I will decouple t_grid_loc_spec and enum GridDefType and move all VIB-related data structures to  the vib_inf.[h|cpp]